### PR TITLE
(feature) #161 Auto Complete Template Text

### PR DESCRIPTION
**Pull request by ycarooliveira**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x]  #Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing

### CODE REVIEW

File: demo/src/ChatContainer.vue [hunk 1]
```diff
@@ -5,36 +5,28 @@
 			<button type="submit" :disabled="disableForm || !addRoomUsername">
 				Create Room
 			</button>
-			<button class="button-cancel" @click="addNewRoom = false">
-				Cancel
-			</button>
+			<button class="button-cancel" @click="addNewRoom = false">Cancel</button>
```
Comment (antoine92190) [inline demo/src/ChatContainer.vue:8]: Please revert to previous code styling

File: demo/src/ChatContainer.vue [hunk 2]
```diff
@@ -5,36 +5,28 @@
...
 
 		<form v-if="inviteRoomId" @submit.prevent="addRoomUser">
 			<input v-model="invitedUsername" type="text" placeholder="Add username" />
 			<button type="submit" :disabled="disableForm || !invitedUsername">
 				Add User
 			</button>
-			<button class="button-cancel" @click="inviteRoomId = null">
-				Cancel
-			</button>
+			<button class="button-cancel" @click="inviteRoomId = null">Cancel</button>
```
Comment (antoine92190) [inline demo/src/ChatContainer.vue:16]: Please revert to previous code styling

File: demo/src/ChatContainer.vue [hunk 3]
```diff
@@ -5,36 +5,28 @@
...
 		</form>
 
 		<form v-if="removeRoomId" @submit.prevent="deleteRoomUser">
 			<select v-model="removeUserId">
-				<option default value="">
-					Select User
-				</option>
+				<option default value="">Select User</option>
```
Comment (antoine92190) [inline demo/src/ChatContainer.vue:21]: Please revert to previous code styling

File: demo/src/ChatContainer.vue [hunk 4]
```diff
@@ -5,36 +5,28 @@
...
 					{{ user.username }}
 				</option>
 			</select>
 			<button type="submit" :disabled="disableForm || !removeUserId">
 				Remove User
 			</button>
-			<button class="button-cancel" @click="removeRoomId = null">
-				Cancel
-			</button>
+			<button class="button-cancel" @click="removeRoomId = null">Cancel</button>
```
Comment (antoine92190) [inline demo/src/ChatContainer.vue:29]: Please revert to previous code styling

File: demo/src/ChatContainer.vue [hunk 5]
```diff
@@ -502,9 +509,7 @@ export default {
 				newMessage.files = deleteDbField
 			}
 
-			await messagesRef(roomId)
-				.doc(messageId)
-				.update(newMessage)
+			await messagesRef(roomId).doc(messageId).update(newMessage)
```
Comment (antoine92190) [inline demo/src/ChatContainer.vue:512]: Please revert to previous code styling

File: demo/src/ChatContainer.vue [hunk 6]
```diff
@@ -516,9 +521,7 @@ export default {
 		},
 
 		async deleteMessage({ message, roomId }) {
-			await messagesRef(roomId)
-				.doc(message._id)
-				.update({ deleted: new Date() })
+			await messagesRef(roomId).doc(message._id).update({ deleted: new Date() })
```
Comment (antoine92190) [inline demo/src/ChatContainer.vue:524]: Please revert to previous code styling

File: demo/src/ChatContainer.vue [hunk 7]
```diff
@@ -548,9 +551,7 @@ export default {
 			await uploadFileRef.put(file.blob, { contentType: type })
 			const url = await uploadFileRef.getDownloadURL()
 
-			const messageDoc = await messagesRef(roomId)
-				.doc(messageId)
-				.get()
+			const messageDoc = await messagesRef(roomId).doc(messageId).get()
```
Comment (antoine92190) [inline demo/src/ChatContainer.vue:554]: Please revert to previous code styling

File: demo/src/ChatContainer.vue [hunk 8]
```diff
@@ -560,9 +561,7 @@ export default {
 				}
 			})
 
-			await messagesRef(roomId)
-				.doc(messageId)
-				.update({ files })
+			await messagesRef(roomId).doc(messageId).update({ files })
```
Comment (antoine92190) [inline demo/src/ChatContainer.vue:564]: Please revert to previous code styling

File: src/lib/Room/Room.vue [hunk 9]
```diff
@@ -298,16 +312,17 @@ import RoomFiles from './RoomFiles/RoomFiles'
...
-import filteredUsers from '../../utils/filter-items'
+import filteredItems from '../../utils/filter-items'
 import Recorder from '../../utils/recorder'
 
 const { detectMobile, iOSDevice } = require('../../utils/mobile-detection')
 
 const debounce = (func, delay) => {
 	let inDebounce
-	return function() {
+	return function () {
```
Comment (antoine92190) [inline src/lib/Room/Room.vue:325]: Please revert to previous code styling

File: src/lib/Room/Room.vue [hunk 10]
```diff
@@ -897,7 +969,7 @@ export default {
 				setTimeout(() => element.classList.remove('vac-scroll-smooth'))
 			}, 50)
 		},
-		onChangeInput: debounce(function(e) {
+		onChangeInput: debounce(function (e) {
```
Comment (antoine92190) [inline src/lib/Room/Room.vue:972]: Please revert to previous code styling

File: src/lib/Room/RoomTemplatesText/RoomTemplatesText.vue [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+				this.activeItem < this.filteredTemplatesText.length - 1
+			) {
+				this.activeItem++
+			} else if (this.activeUpOrDown < 0 && this.activeItem > 0) {
+				this.activeItem--
+			}
+			this.$emit('active-item')
+		}
+	},
+	methods: {}
```
Comment (antoine92190) [inline src/lib/Room/RoomTemplatesText/RoomTemplatesText.vue:67]: You can remove empty `methods`

**Comment by antoine92190** [comment]
Thanks a lot for this feature!
I will test it locally, but in the meantime could you revert the code styling changes above?
I'm using `prettier` and `eslint` to enforce code styling convention.

**Comment by ycarooliveira** [comment]
Nah I should have send this PR a long time ago...
I'll try to revert the styling, my `vscode` is acting up, it's ignoring the `eslint` file for some reason and using my default config settings

**Comment by antoine92190** [comment]
@ycarooliveira Why did you close this?

**Comment by ycarooliveira** [comment]
> @ycarooliveira Why did you close this?

err Idk LMAO

**Comment by antoine92190** [comment]
Thanks again! Love the UX to select templates with keyboard, will add that for user tags and emojis :)

**Pull request merged**